### PR TITLE
Input: Handle window focus loss. Fix runaway mouse axes

### PR DESCRIPTION
**Pull request by RobotCaleb**
This handles window focus loss. We don't process any input while focus is lost.
Current state is responsible for handling focus changes.

Also addresses bug with SDL_SetRelativeMouseMode by clamping mouse axis per event.

### CODE REVIEW

**Comment by chrisforbes** [comment]
I don't like this. Please continue releasing the mouse grab when in the menus.

File: main.cc [hunk 1]
```diff
@@ -1444,14 +1450,32 @@ run()
...
+                }
 
             case SDL_MOUSEMOTION:
-                mouse_axes[EN_MOUSE_AXIS(input_mouse_x)] += e.motion.xrel;
-                mouse_axes[EN_MOUSE_AXIS(input_mouse_y)] += e.motion.yrel;
+            {
+                auto x = e.motion.xrel;
+                auto y = e.motion.yrel;
+
+                x = x > 0 ? std::min(MAX_AXIS_PER_EVENT, x) : std::max(-MAX_AXIS_PER_EVENT, x);
```
Comment (chrisforbes) [inline main.cc]: Let's have a `clamp()` that does this rather than the `?:`

Comment (chrisforbes) [inline main.cc]: (It would be ideal to write this as:

`auto x = clamp(e.motion.xrel, -MAX_AXIS_PER_EVENT, MAX_AXIS_PER_EVENT);`

**Comment by RobotCaleb** [comment]
"I don't like this. Please continue releasing the mouse grab when in the menus."
Not sure what this means. We do release the mouse in menus.

**Comment by chrisforbes** [comment]
Apologies. I misread, and had not actually run the code.

**Comment by RobotCaleb** [comment]
Clamped.
Do test, we may need to adjust the max axis value. I only know my system never sent a dt greater than abs 128 except when in the bad state the value exists to remove.

**Comment by chrisforbes** [comment]
This works OK in my testing. Merging.

**Pull request merged**